src/utils/AdaptiveTimeStep.cpp

```diff
@@ -239,16 +225,87 @@ AdaptiveTimeStep::Calculate (
     }
 }
 
+void
+AdaptiveTimeStep::GatherMinAccSlice (MultiBeam& beams, const amrex::Geometry& geom,
+                                     const Fields& fields, const int islice)
+{
+    using namespace amrex::literals;
+
+    if (m_do_adaptive_time_step == 0) return;
+
+    HIPACE_PROFILE("AdaptiveTimeStep::Calculate()");
+
+    const PhysConst phys_const = get_phys_const();
+    const amrex::Real clightinv = 1._rt/phys_const.c;
+
+    constexpr int lev = 0;
+
+    // Extract properties associated with physical size of the box
+    const int nbeams = beams.get_nbeams();
+    const int numprocs_z = Hipace::GetInstance().m_numprocs_z;
+
+    for (int ibeam = 0; ibeam < nbeams; ibeam++) {
+        if (!Hipace::HeadRank()) break;
```

Comment:
```suggestion
```
I think this doesn’t belong here anymore, or does only the headrank need `MinAcc`?

Reply:
Correct, I took the occasion to do some more cleaning.